CantoboardFramework/Keyboard/CandidateOrganizer.swift

```diff
@@ -82,15 +82,21 @@ class InputEngineCandidateSource: CandidateSource {
             candidatePaths.append([])
         }
         
+        let rawInputWithoutDigits = inputEngine.englishComposition?.text.lowercased() ?? ""
+        var demotedEnglishCandidateIndices: [Int] = []
+        
         // If input is an English word, insert best English candidates first.
         if !hasPopulatedPrefectEnglishCandidates && isEnglishActive {
             for i in 0..<inputEngine.englishPrefectCandidatesStartIndex {
-                candidatePaths[0].append(CandidatePath(source: .english, index: i))
+                if inputMode == .english || englishCandidates[i].lowercased() != rawInputWithoutDigits.lowercased() {
```

Comment:
Would you mind giving some example how this changes the candidates?

Reply:
This is only to demote the candidate(s) same as what the user has typed in Chinese mode. You don’t what “man” to be at the very front when you type “man”.

Reply:
Or it would be better to put the word after the 20th candidate, maybe, instead of at the very bottom.
But we should only do so for high-frequency English words, but not an arbitrary “word” such as “ga”.